readthedocs/core/static-src/core/js/doc-embed/search.js

```diff
@@ -23,48 +23,48 @@ function attach_elastic_search_query(data) {
 
         search_url.href = api_host;
         search_url.pathname = '/api/v2/docsearch/';
-        search_url.search = '?q=' + $.urlencode(query) + '&project=' + project +
-            '&version=' + version + '&language=' + language;
+        search_url.search = '?query=' + $.urlencode(query) + '&project=' + project +
+                            '&version=' + version + '&language=' + language;
 
         search_def
-            .then(function (results) {
-                var hits = results.hits || {};
-                var hit_list = hits.hits || [];
+            .then(function (data) {
+                var hit_list = data.results || [];
+                var total_count = data.count || 0;
 
                 if (hit_list.length) {
-                    for (var n in hit_list) {
-                        var hit = hit_list[n];
-                        var fields = hit.fields || {};
-                        var list_item = $('<li style="display: none;"></li>');
-                        var item_url = document.createElement('a');
-                        var highlight = hit.highlight;
-
-                        item_url.href += fields.link +
-                            DOCUMENTATION_OPTIONS.FILE_SUFFIX;
-                        item_url.search = '?highlight=' + $.urlencode(query);
-
-                        // Result list elements
-                        list_item.append(
-                            $('<a />')
-                            .attr('href', item_url)
-                            .html(fields.title)
-                        );
-                        // fields.project is returned as an array
-                        if (fields.project.indexOf(project) === -1) {
-                            list_item.append(
-                                $('<span>')
-                                .text(" (from project " + fields.project + ")")
-                            );
+                    for (var i = 0; i < hit_list.length; i += 1) {
+                        var doc = hit_list[i];
+                        var highlight = doc.highlight;
+                        var $list_item = $('<li style="display: none;"></li>');
+
+                        // Creating the result from elements
+                        var link = doc.link + DOCUMENTATION_OPTIONS.FILE_SUFFIX +
+                                   '?highlight=' + $.urlencode(query);
+
+                        var $item = $('<a>', {'href': link});
+                        $item.html(doc.title);
+                        $list_item.append($item);
+
+                        // If the document is from subproject, add extra information
+                        if (doc.project !== project) {
```

Comment:
Be careful that this doesn't revert to buggy behavior, if the return attribute`doc.proejct` hasn't changed. The equivalent of `doc.project` in the current js is always an array, so `doc.project !== project` is false because `doc.project = ['docs']` and `project = 'docs'`

Reply:
As discussed over slack, the new API always return `String`, instead of list.